### PR TITLE
1322 use a3s as authorisation server -> Master

### PR DESCRIPTION
**Pull request by VictorSzewczenko**
This PR does the following:

- Configures Dokuti to run with A3S as the authorisation server. Updates the default internal application settings to run against a locally running A3S server with a public key that matches the development generated tempRSA key material for the identity server.
- Removes the access token customiser class, as it was designed for use with Keycloak tokens. The default Springboot token extraction implementation works for the more standard IDS4 tokens.
- Updated the security context util to be able to extract permissions from tokens.
- Added a Service that is intended to be used as the permissions assessor within the @PreAuthorize annotation.

### CODE REVIEW

File: development-resources/docker-compose/dokuti-development/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: '3'
```
Comment (melissapalmer) [inline development-resources/docker-compose/dokuti-development/docker-compose.yml]: Should you update the docker-compose file for /quickstart/docker-compose too?

Comment (VictorSzewczenko) [reply development-resources/docker-compose/dokuti-development/docker-compose.yml]: Agreed, will definitely do that now.

**Comment by melissapalmer** [comment]
@VictorSzewczenko have also just picked up that you are going to need to update the README such that it does not refer to `Credentials for Keycloak`

**Comment by VictorSzewczenko** [comment]
> @VictorSzewczenko have also just picked up that you are going to need to update the README such that it does not refer to `Credentials for Keycloak`

@melissapalmer Agreed. I have not actually updated it yet. Im actually going to either close this PR or update to read WIP, as it is currently not yet ready for merge. Essentially, A3S needs to be open sourced before this repo can be updated to use it as a dependency. Once that is done, then the actual complete replacement can be concluded.

**Comment by VictorSzewczenko** [comment]
Closing this PR, as the complete replacement of Keycloak with A3S can only be done once A3S is open sourced. The work in this branch will be concluded once that is done.

**Pull request closed without merging**